app/collectors/bucket.scala

```diff
@@ -43,23 +46,23 @@ object Bucket {
 
   private def arn(bucketName: String) = s"arn:aws:s3:::$bucketName" 
 
-  def fromApiData(bucket: AWSBucket, client: AmazonS3): Option[Bucket] = {
-    val bucketName = bucket.getName
+  def fromApiData(bucket: AWSBucket, client: S3Client, originRegion: Region): Option[Bucket] = {
+    val bucketName = bucket.name
     try {
-      val bucketRegion = client.getBucketLocation(bucket.getName)
-      if (bucketRegion == client.getRegionName) {
+      val bucketRegion = client.getBucketLocation(GetBucketLocationRequest.builder().bucket(bucketName).build()).locationConstraintAsString()
+      if (bucketRegion == originRegion.toString) {
```

Comment:
I'd prefer using `id` here. As it happens `toString` is the same and will work, but `id` would be cleaner and more explicit.
```suggestion
      if (bucketRegion == originRegion.id) {
```